firebase-installations/src/main/java/com/google/firebase/installations/GetIdListener.java

```diff
@@ -27,15 +27,16 @@ public GetIdListener(TaskCompletionSource<String> taskCompletionSource) {
   @Override
   public boolean onStateReached(PersistedFidEntry persistedFidEntry) {
     if (persistedFidEntry.isUnregistered() || persistedFidEntry.isRegistered()) {
-      taskCompletionSource.setResult(persistedFidEntry.getFirebaseInstallationId());
+      taskCompletionSource.trySetResult(persistedFidEntry.getFirebaseInstallationId());
       return true;
     }
+    return false;
+  }
 
+  @Override
+  public boolean onException(PersistedFidEntry persistedFidEntry, Exception exception) {
     if (persistedFidEntry.isErrored()) {
-      taskCompletionSource.setException(
-          new FirebaseInstallationsException(
-              "Failed to update client side cache.",
-              FirebaseInstallationsException.Status.CLIENT_ERROR));
+      taskCompletionSource.trySetException(exception);
```

Comment:
given this is the same in each of the listener impls so far we could move it into a default interface method

Reply:
error: [NoStaticOrDefaultMethodsInInterfaces] Avoid static/default methods in interfaces: We currently desugar SDKs with retrolambda which has limited support for those.

Default methods require API level 24. Current min is 14

Reply:
oh, sad! I thought we had moved past retrolambda